// opengpio/dist/cjs/devices/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Default = exports.Radxa_Rock_S0 = exports.OrangePi_5 = exports.RaspberryPi_Zero2W = exports.RaspberryPi_ZeroW = exports.RaspberryPi_5B = exports.RaspberryPi_400 = exports.RaspberryPi_4B = exports.RaspberryPi_3BPlus = exports.RaspberryPi_3B = exports.RaspberryPi_2B = exports.NanoPi_NEO3 = void 0;
var NanoPi_NEO3_1 = require("./NanoPi_NEO3");
Object.defineProperty(exports, "NanoPi_NEO3", { enumerable: true, get: function () { return NanoPi_NEO3_1.NanoPi_NEO3; } });
var RaspberryPi_2B_1 = require("./RaspberryPi_2B");
Object.defineProperty(exports, "RaspberryPi_2B", { enumerable: true, get: function () { return RaspberryPi_2B_1.RaspberryPi_2B; } });
var RaspberryPi_3B_1 = require("./RaspberryPi_3B");
Object.defineProperty(exports, "RaspberryPi_3B", { enumerable: true, get: function () { return RaspberryPi_3B_1.RaspberryPi_3B; } });
var RaspberryPi_3BPlus_1 = require("./RaspberryPi_3BPlus");
Object.defineProperty(exports, "RaspberryPi_3BPlus", { enumerable: true, get: function () { return RaspberryPi_3BPlus_1.RaspberryPi_3BPlus; } });
var RaspberryPi_4B_1 = require("./RaspberryPi_4B");
Object.defineProperty(exports, "RaspberryPi_4B", { enumerable: true, get: function () { return RaspberryPi_4B_1.RaspberryPi_4B; } });
var RaspberryPi_400_1 = require("./RaspberryPi_400");
Object.defineProperty(exports, "RaspberryPi_400", { enumerable: true, get: function () { return RaspberryPi_400_1.RaspberryPi_400; } });
var RaspberryPi_5B_1 = require("./RaspberryPi_5B");
Object.defineProperty(exports, "RaspberryPi_5B", { enumerable: true, get: function () { return RaspberryPi_5B_1.RaspberryPi_5B; } });
var RaspberryPi_ZeroW_1 = require("./RaspberryPi_ZeroW");
Object.defineProperty(exports, "RaspberryPi_ZeroW", { enumerable: true, get: function () { return RaspberryPi_ZeroW_1.RaspberryPi_ZeroW; } });
var RaspberryPi_Zero2W_1 = require("./RaspberryPi_Zero2W");
Object.defineProperty(exports, "RaspberryPi_Zero2W", { enumerable: true, get: function () { return RaspberryPi_Zero2W_1.RaspberryPi_Zero2W; } });
var OrangePi_5_1 = require("./OrangePi_5");
Object.defineProperty(exports, "OrangePi_5", { enumerable: true, get: function () { return OrangePi_5_1.OrangePi_5; } });
var Radxa_Rock_S0_1 = require("./Radxa_Rock_S0");
Object.defineProperty(exports, "Radxa_Rock_S0", { enumerable: true, get: function () { return Radxa_Rock_S0_1.Radxa_Rock_S0; } });
var Default_1 = require("./Default");
Object.defineProperty(exports, "Default", { enumerable: true, get: function () { return Default_1.Default; } });
